### PR TITLE
Add dropdown to select token when creating a new task

### LINKED ISSUE #14020: User selects token for task
In Data Explorer Save As:

Given I am a user creating a task and a list of my existing tokens
When I create a task I can select a token from a dropdown
Then the created task will be run with that token

### PR DESCRIPTION
**Pull request by Palakp41**
Closes #14020 

Describe your proposed changes here.
Added token dropdown on task form
Added redux state to store the selected token
Updated edit task page to show token dropdown.
Updated task form in data explorer to include a token dropdown

- [ ] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [ ] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [ ] Rebased/mergeable
- [ ] Tests pass

### CODE REVIEW

File: ui/src/dataExplorer/components/SaveAsTaskForm.tsx [hunk 1]
```diff
@@ -65,6 +74,8 @@ class SaveAsTaskForm extends PureComponent<Props & WithRouterProps> {
     })
 
     setNewScript(this.activeScript)
+    await this.props.getTokens()
+    this.props.setTaskToken(this.props.tokens[0])
```
Comment (alexpaxton) [inline ui/src/dataExplorer/components/SaveAsTaskForm.tsx]: Does this need a guard?

File: ui/src/tasks/components/TaskForm.tsx [hunk 2]
```diff
@@ -145,6 +153,24 @@ export default class TaskForm extends PureComponent<Props, State> {
       </Form>
     )
   }
+  private get tokenDropdown(): JSX.Element {
```
Comment (alexpaxton) [inline ui/src/tasks/components/TaskForm.tsx]: I can hear Brandon's voice in my head saying "if you have a getter that returns a component it should probably be it's own component"

File: ui/src/dataExplorer/components/SaveAsTaskForm.tsx (hunk 1 above)
Comment (Palakp41) [reply ui/src/dataExplorer/components/SaveAsTaskForm.tsx]: Yes, I updated it

File: ui/src/tasks/components/TaskForm.tsx (hunk 2 above)
Comment (Palakp41) [reply ui/src/tasks/components/TaskForm.tsx]: I agree. I created it to be separate component

**Pull request merged**